hardware/1Wire.cpp

```diff
@@ -32,57 +32,44 @@ C1Wire::C1Wire(const int ID, const int sensorThreadPeriod, const int switchThrea
 {
 	m_HwdID = ID;
 
-	// Defaults for pre-existing 1wire instances
-	if (0 == m_sensorThreadPeriod)
-		m_sensorThreadPeriod = 5 * 60 * 1000;
-
-	if (0 == m_switchThreadPeriod)
-			m_switchThreadPeriod = 100;
-
 	DetectSystem();
 }
 
 C1Wire::~C1Wire()
 {
 }
 
-bool C1Wire::Have1WireSystem()
-{
-#ifdef WIN32
-	return (C1WireForWindows::IsAvailable());
-#else // WIN32
-	return true;
-#endif // WIN32
-}
-
 void C1Wire::DetectSystem()
 {
-#ifdef WIN32
-	if (!m_system && C1WireForWindows::IsAvailable())
-		m_system=new C1WireForWindows();
-#else // WIN32
-
-	// Using the both systems at same time results in conflicts,
-	// see http://owfs.org/index.php?page=w1-project.
-	if (m_path.length() != 0) {
-		m_system=new C1WireByOWFS(m_path);
-	} else if (C1WireByKernel::IsAvailable()) {
-		m_system=new C1WireByKernel();
-	} else {
-		m_system=new C1WireByOWFS(m_path);
+	if (m_path.length() != 0) 
+	{
+		m_system = new C1WireByOWFS(m_path);
 	}
-
-#endif // WIN32
+	else
+	{
+		#ifdef WIN32
+		if (!m_system && C1WireForWindows::IsAvailable())
+			m_system = new C1WireForWindows();
+		#else // WIN32
+		// Using the both systems at same time results in conflicts,
+		// see http://owfs.org/index.php?page=w1-project.
+		if (C1WireByKernel::IsAvailable()) 
+			m_system=new C1WireByKernel();
+		#endif // WIN32
+	 }
 }
 
 bool C1Wire::StartHardware()
 {
+	if (m_system == NULL)
+		return false;
+
 	// Start worker thread
-	if (0 != m_sensorThreadPeriod)
+	if (-1 != m_sensorThreadPeriod)
```

Comment:
This is not true.
I c++ and on many other languages, methods return -1 as false.
Many times "0" means something (lenght etc).